Remove qchem deprecated functionality for fermionic observables (#4556)
PennyLaneAI/pennylane: doc/development/deprecations.rst

  - Deprecated in v0.31

* The ``QuantumScript.set_parameters`` method and the ``QuantumScript.data`` setter has
  been deprecated. Please use ``QuantumScript.bind_new_parameters`` instead.

  - Deprecated in v0.32
  - Will be removed in v0.33

Completed deprecation cycles
----------------------------

> * ``qml.qchem.jordan_wigner`` is deprecated, and usage will now raise a warning.

Reviewer: ```suggestion
* ``qml.qchem.jordan_wigner`` had been removed.
```

  Use ``qml.jordan_wigner`` instead. List input to define the fermionic operator

>   is also deprecated; the fermionic operators ``qml.FermiA``, ``qml.FermiC``,

Reviewer: ```suggestion
 is no longer accepted; the fermionic operators ``qml.FermiA``, ``qml.FermiC``,
```
